authedclient.go

```diff
@@ -0,0 +1,140 @@
+package gomatrixserverlib
+
+import (
+	"encoding/json"
+	"github.com/matrix-org/gomatrix"
+	"golang.org/x/crypto/ed25519"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+// An AuthedClient is a matrix federation client that adds
+// "Authorization: X-Matrix" headers to requests that need ed25519 signatures
+type AuthedClient struct {
```

Comment:
This is explicitly only relevant for federation and yet we don't mention "federation" in the type, but do randomly mention "auth". AFAICT, *every* request made by this client has X-Matrix headers added, so mentioning a random impl detail in the type feels strange. Can't we just call this `FederationClient`?